pex/pip/installation.py

```diff
@@ -82,19 +88,61 @@ def _vendored_installation(interpreter=None):
     )
 
 
+def _bootstrap_pip(
```

Comment:
This is the new trick that allows a Pip that supports Python 3.12 to be bootstrapped without resorting to a resolve using the vendored Pip (the technique on line 142 below). Not much of a trick, pretty obvious in hindsight.